contrib/pyln-client/pyln/client/gossmap.py

```diff
@@ -125,7 +125,12 @@ def _update_channel(self,
                                   fields.get('htlc_maximum_msat', None),
                                   fields['fee_base_msat'],
                                   fields['fee_proportional_millionths'])
-        self.half_channels[direction] = half
+        if (self.half_channels[direction] is None):
+            self.half_channels[direction] = half
+        # gossip_store contains both broadcastable and routing-only updates
+        # Check that we're not replacing a newer update with broadcastable
+        elif (half.timestamp > self.half_channels[direction].timestamp):
+            self.half_channels[direction] = half
```

Comment:
I think we should filter when we enter the gossip store (if we don't already?) such that later messages will always have greater timestamp than earlier ones?  So this logic is simply "override previous".

Reply:
That would normally be the case.  I was concerned with an edge case where we have a normal message with timestamp 1, then a rate-limited message with timestamp 3.  Some time later, our rate-limit clears and we receive a message with timestamp 2.  This could replace the original broadcastable message, but would not supersede the later-timestamped gossip that's still useful for the routing graph.
This edge case wouldn't normally occur if we're well connected and the originating node is always incrementing gossip timestamps, but it is a valid case.  We could ignore messages in this edge case and keep the gossip_store in ascending order.  Alternatively, we could discard message 2 and unflag message 3 (now okay to broadcast also.) Would probably be more useful to delete and rewrite it to the store without the spam flag, but same idea.  I kind of like this behavior now that I think about it.

Reply:
Yes!  We should be comparing any incoming against the *latest we have*, spam or no.

Don't worry too much about the case where we might consider something "no longer spam" sometime later, either.